wikipedia_cleanup/property_correlation.py

```diff
@@ -84,22 +91,60 @@ def _sparse_time_series_conversion(
     def _load_cache(self, possible_cached_mapping: Path) -> bool:
         try:
             if possible_cached_mapping.exists():
-                print(f"Cached model found, loading from {possible_cached_mapping}")
                 with open(possible_cached_mapping, "rb") as f:
-                    self.related_properties_lookup = pickle.load(f)
+                    cache = pickle.load(f)
+                    if (
+                        cache["num_required_changes"] != self.NUM_REQUIRED_CHANGES
+                        or cache["max_allowed_properties"]
+                        != self.MAX_ALLOWED_PROPERTIES
+                        or cache["percent_allowed_mismatch"]
+                        < self.PERCENT_ALLOWED_MISMATCHES
+                    ):
+                        return False
+
+                    print(f"Cached model found, loading from {possible_cached_mapping}")
+                    self.related_properties_lookup = cache["related_properties_lookup"]
                 return True
             else:
                 print("No cache found, recalculating model.")
         except EOFError:
             print("Caching failed, recalculating model.")
         return False
 
+    def _save_cache(self, possible_cached_mapping):
+        possible_cached_mapping.parent.mkdir(exist_ok=True, parents=True)
+        print(f"Saving cache to {possible_cached_mapping.name}.")
+        with open(possible_cached_mapping, "wb") as f:
+            cache = {
+                "num_required_changes": self.NUM_REQUIRED_CHANGES,
+                "max_allowed_properties": self.MAX_ALLOWED_PROPERTIES,
+                "percent_allowed_mismatch": self.PERCENT_ALLOWED_MISMATCHES,
+                "related_properties_lookup": self.related_properties_lookup,
+            }
+            pickle.dump(cache, f)
+
+    def _prepare_related_properties_lookup_for_inference(self):
+        for (
+            curr_property_key,
+            related_properties,
+        ) in self.related_properties_lookup.items():
+            filtered_related_properties_without_distances = []
+            for related_property_key, related_property_distance in related_properties:
+                if related_property_distance <= self.PERCENT_ALLOWED_MISMATCHES:
```

Comment:
The cache also saves `percent_allowed_mismatches` so this method would only need to be executed if thats actually unequal to the current `self.PERCENT_ALLOWED_MISMATCHES`

Reply:
That's not true since it changes the format of self.related_properties_lookup changes by removing the percentage.